Enhance the existing docs for integrating Quarkus with Azure  (apply style, diataxis, new conceptual information)

michelle-purcell opened the pull request:
This PR aims to enhance the existing Quarkus docs for Azure integration as follows:

- Iteration 1 of the restructure into Diataxis content types. Depending on SME input, more restructuring might be needed later. _(Left file renaming & URL redirection until later... after initial input.)_
- Edits to enhance and style the 4 existing guides as per the [Quarkus doc contributor guide](https://quarkus.io/guides/doc-reference).
- Added a new concept topic [Quarkus integration with Microsoft Azure](https://quarkus-pr-main-32823-preview.surge.sh/version/main/guides/azure-integration-built-in-concept), which aims to:
  -  Explain and introduce Quarkus integration with Microsoft Azure and link to other valuable Quarkus docs
  -  Links to existing valuable tutorial resources in Quarkus docs, the Microsoft website, and the superheroes workshops.
 - Added a new reference topic: [Quarkus extensions for Azure integration](https://quarkus-pr-main-32823-preview.surge.sh/version/main/guides/azure-extensions-reference)

- Fixes #32441

maxandersen commented on docs/src/main/asciidoc/azure-support-overview-concept.adoc:
```diff
@@ -0,0 +1,117 @@
+
+link:https://learn.microsoft.com/en-us/azure/?product=popular[Microsoft Azure] is a versatile and flexible cloud platform, which you can use to host your Quarkus applications.
+Azure supports various programming languages, complex architectures and frameworks, including Java and Quarkus.
+Azure also fully supports the Apache Maven Java development tool.
```
Not sure what this refers to. 
Makes it sounds like you can use it with gradle? I assume that is not that case.

michelle-purcell replied:
Fixed. Removed the 'marketing spiel' and restructured the new integration overview concept topic.

maxandersen commented on docs/src/main/asciidoc/azure-support-overview-concept.adoc:
```diff
@@ -0,0 +1,117 @@
+
+== Quarkus extensions for Microsoft Azure
+
+Quarkus provides the following extensions to make it easier to integrate your existing Quarkus applications with Microsoft Azure:
```
These two are different from the others in that it is part of funky which is abstraction available in quarkus that lets you use serverless function style programming while mapping it to jaxrs existing programming model. Ie. Allowing to be portable. 

Azure functions is one of the supported modes. Amazon lambda is another.

michelle-purcell replied:
I created a new reference topic [Quarkus extensions for Azure integration](https://quarkus-pr-main-32823-preview.surge.sh/version/main/guides/azure-extensions-reference)  to provide guidance on the extensions and links to any existing resources we have that explain a bit about when and how to use them.

maxandersen commented on docs/src/main/asciidoc/azure-support-overview-concept.adoc:
```diff
@@ -0,0 +1,117 @@
+* `quarkus-azure-functions`
+* `quarkus-azure-functions-http`
+
+// What about these extensions? Should we mention them?
```
These are different from the funky ones in the sense these are providing integration with specific azure cloud services. 

And yes; if this guide is about what you can do with quarkus in azure it makes sense to mention them.

maxandersen commented on docs/src/main/asciidoc/azure-support-overview-concept.adoc:
```diff
@@ -0,0 +1,117 @@
+
+=== Microsoft Azure deployment tools
+
+You can run Quarkus applications on Azure by deploying them as a Docker container and  using a combination of some of the following Microsoft integration tools:
```
Hmm - we shouldn't need to push azure specific features first. 

Shouldn't we show the ways quarkus can deploy that requires zero or very little setup first ?

Ie. Deploying to Azure container service should be doable 100% with our builtin quarkus features.

michelle-purcell replied:
@maxandersen - Thanks very much Max for the initial feedback. This is really helpful :+1:  Sorry for the slow reply. I was on PTO. 

I'll rework the content, based on your initial feedback above :arrow_up: 

The aim of the concept topic was to provide some context to some of the other existing doc resources that are available for deploying and developing on Azure, which will also be restructured into diatazis content types as part of this issue. If I understand correctly, we need to make the distinction between deploying info and developing/integrating services with Azure Functions. Would it make more sense to have 2 separate concept pages to introduce Quarks deployment on Azure versus Quarkus development/integration with Azure Functions? Thanks

maxandersen reviewed (changes requested):
This reads more like what we would put on a website page on quarkus.io to give overview on azure quarkus support.  

That is actually something we have on roadmap in form of a quarkus.io/cloud page with subpages for this content. 

I would say we put something g like there over a guide.

michelle-purcell committed:
Diataxis & Quarkus styling of existing Azure docs

Fix link

Fix vale errors

Added new reference topic and updated links

ID fix

fix

enhancements

Fix draft

Add categories for findability

Fix numbering

Fixed title

Update docs/src/main/asciidoc/azure-functions-http.adoc

Co-authored-by: Erin Schnabel <ebullientworks@gmail.com>

Fix SME comments from Erin

Fix SME comments from Erin

Fixed conflicts and applied further edits

Fixed conflicts and more edits

Further edits

Fix duplication

Fix typo

Fixed SME comments (danieloh30)

Fix SME comments

Hide content referring to Gradle until supported

Minor fixes following diataxis updates

ass missing attribute

ebullient commented on docs/src/main/asciidoc/azure-functions-http.adoc:
```diff
@@ -50,68 +71,92 @@ If you don't log in to Azure you won't be able to deploy.
 [source,bash,subs=attributes+]
 ----
 ./mvnw clean package quarkus:dev
 ----
```
Replace this w/ include of dev mode snippet?

```
include::{includes}/devtools/dev.adoc[]
```

michelle-purcell replied:
Fixed.

ebullient commented on docs/src/main/asciidoc/azure-functions-http.adoc:
```diff
@@ -50,68 +71,92 @@ If you don't log in to Azure you won't be able to deploy.
+Congratulations!
+You have learned how to create, build, and deploy a sample Quarkus RESTEasy microservice to Azure Functions.
 
+// Should this step be positioned earlier in this guide? It seems a bit late here...
```
If/when we split, this is something for the reference guide, rather than the tutorial..

ebullient commented on docs/src/main/asciidoc/azure-functions-http.adoc:
```diff
@@ -3,45 +3,66 @@ This guide is maintained in the main Quarkus repository
 [source,bash,subs=attributes+]
 ----
 quarkus create app --extension=quarkus-azure-functions-http
 ----
```
Replace with project create snippet?

```
:create-app-extensions: azure-functions-http
include::{includes}/devtools/create-app.adoc[]
```

michelle-purcell replied:
Fixed.

ebullient commented on docs/src/main/asciidoc/azure-functions.adoc:
```diff
@@ -57,35 +80,44 @@ public class Function {
 ----
 quarkus create app --extension=quarkus-azure-functions
 ----
 
```
Use includes instead?

```
:create-app-extensions: azure-functions
include::{includes}/devtools/create-app.adoc[]
```

michelle-purcell replied:
Fixed.

michelle-purcell commented:
Thanks for your review @ebullient :+1:  I updated the content as per your suggestions.

michelle-purcell commented:
@patriot1burke - I would appreciate your review of the Azure Functions content in this PR. I applied some Quarkus-style edits to several topics that you also recently worked on. Thanks

michelle-purcell commented on docs/src/main/asciidoc/azure-integration-built-in.adoc:
```diff
@@ -0,0 +1,102 @@
+
+Quarkus can be integrated with developer tools and managed services that are available on the Microsoft Azure cloud platform.
+
+You can deploy and host your containerized Quarkus applications to Microsoft Azure by using link:https://learn.microsoft.com/en-us/azure/container-apps/[Azure Container Apps] and then push them to the link:https://azure.microsoft.com/en-us/products/app-service/[Azure App Service] where they can run in a scalable Java runtime.
```
No not yet. We link to the Microsoft document. If we need to document this in the Quarkus doc portal, we need to create an issue to add to the backlog and plan. Thanks.

michelle-purcell commented on docs/src/main/asciidoc/azure-integration-built-in.adoc:
```diff
@@ -0,0 +1,102 @@
+:categories: cloud,integration
+
+Quarkus provides built-in extensions for integrating your containerized Java applications with Microsoft Azure.
+You can use the cloud-hosting capabilities of Azure to run and scale your Quarkus applications in a Java runtime on the Azure App Service.
```
No. But we link to this one: 

To deploy your Quarkus native container image to Azure App Service, see the following tutorial:

[Tutorial: Build a custom image and run in App Service from a private registry](https://docs.microsoft.com/en-us/azure/app-service/containers/tutorial-custom-docker-image?WT.mc_id=opensource-quarkus-brborges*)

michelle-purcell commented on docs/src/main/asciidoc/azure-extensions.adoc:
```diff
@@ -0,0 +1,56 @@
+`quarkus-funqy-http`
+|Write and deploy serverless Quarkus applications that use link:https://quarkus.io/guides/funqy[Quarkus Funqy HTTP binding] with Azure functions.
+
+link:https://quarkus.io/guides/funqy[Quarkus Funqy] provides a portable Java API to write functions, mapped to the JAX-RS existing programming model.
```
Fixed.

michelle-purcell commented on docs/src/main/asciidoc/azure-extensions.adoc:
```diff
@@ -0,0 +1,56 @@
+For example, you can write microservices with RESTEasy Reactive (our Jakarta REST implementation), Undertow (servlet), Reactive Routes, or Funqy HTTP and to make these microservices deployable to the link:https://learn.microsoft.com/en-us/azure/azure-functions/[Azure Functions] runtime.
+
+The `quarkus-azure-functions-http` extension also provides support for Azure Functions triggers and bindings, making it easy to integrate your Java code with other Azure services.
+Use when you want your function to be triggered by multiple sources, for example, `HttpTrigger`, `BlobTrigger``, and `CosmosDBTrigger`.
```
Fixed.

michelle-purcell commented on docs/src/main/asciidoc/azure-extensions.adoc:
```diff
@@ -0,0 +1,56 @@
+|Build cloud-native Java applications by using Quarkus and Azure Functions.
+For example, you can write microservices with RESTEasy Reactive (our Jakarta REST implementation), Undertow (servlet), Reactive Routes, or Funqy HTTP and to make these microservices deployable to the link:https://learn.microsoft.com/en-us/azure/azure-functions/[Azure Functions] runtime.
+
+The `quarkus-azure-functions-http` extension also provides support for Azure Functions triggers and bindings, making it easy to integrate your Java code with other Azure services.
```
Fixed.

michelle-purcell commented:
@danieloh30 - Thanks so much for your review and comments. I have fixed or responded to each comment.

If there are new tutorials or docs needed in the Quarkus docs portal as per the comments above, can you create an issue for each item that's needed and we can make a plan to implement them one by one? Thanks.

michelle-purcell commented:
@danieloh30  @tqvarnst & @patriot1burke @maxandersen  - 

Given the discussion in Quarkus GH issue #33928, I have removed the specific instructions that I added last week for Gradle  (by hiding with //) as this is not yet supported for the use cases in this content. 

Please can you complete your review this week while the content is up-to-date with `main`. 
I'd like to get this merged if approved by the end of this week.

Thanks :pray:

michelle-purcell commented:
Last call for passengers boarding flight Q3.2 :flying_saucer: 

@maxandersen , @tqvarnst , @iocanel , @patriot1burke - This PR still needs a review and final approval from an SME.

**Note:** There are 2 new topics in this PR, which are outside the backport criteria so it would be good to get this merged before the final freeze for community 3.2.

michelle-purcell commented:
@maxandersen , @tqvarnst , @iocanel , @patriot1burke: This PR has been open for a long time without final approval. Can anyone help here? Thanks :-)
